plutus-scb/src/Plutus/SCB/Core/ContractInstance.hs

```diff
@@ -143,13 +144,19 @@ processFirstInboxMessage ::
 processFirstInboxMessage instanceID (Last msg) = do
     logInfo $ "processFirstInboxMessage start for " <> render (pretty instanceID)
     logInfo . render $ "The first message is: " <+> pretty msg
-    logInfo "Looking up the contract."
+    logInfo "Looking up current state of the contract instance."
     -- look up contract 't'
     ContractInstanceState{csCurrentIteration, csCurrentState, csContractDefinition} <- lookupContractState @t instanceID
+    logInfo . render $ "Current iteration:" <+> pretty csCurrentIteration
+    when (csCurrentIteration /= rspItID msg) $ do
+        logInfo "The first inbox message does not match the contract instance's iteration."
     when (csCurrentIteration == rspItID msg) $ do
+        logInfo "The first inbox message matches the contract instance's iteration."
```

Comment:
Small thing, but it seems a little weird to do two `when` clauses instead of an `if`.

Reply:
yeah you're right, I'll change that.